client/mysqldump.c

```diff
@@ -742,6 +746,14 @@ static void write_header(FILE *sql_file, char *db_name)
             "/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;\n",
             path?"":"NO_AUTO_VALUE_ON_ZERO",compatible_mode_normal_str[0]==0?"":",",
             compatible_mode_normal_str);
+    fprintf(sql_file,
+            "/*!50601 SELECT count(*) INTO @is_rocksdb_supported FROM"
```

Comment:
I only don't see a point in pushing a known-wrong version marker, but even that is benign as you say. Either patching up it to 5.1 or removing is fine